### PR TITLE
New function for determining optimal WCS/shape for set of images

### PR DESCRIPTION
**Pull request by astrofrog**
This adds a new function for determining the optimal WCS/shape for a mosaic of images (or for a single image if changing coordinate frame). This is an alternative to the implementation of a a similar function in https://github.com/astrofrog/reproject/pull/125, but the version here takes into account different coordinate frames and can automatically rotate images if needed.

What is missing here compared to https://github.com/astrofrog/reproject/pull/125 is that we should use the minimal image excluding NaNs as input rather than the full images.

### CODE REVIEW

**Comment by astrofrog** [comment]
@keflavich @mhardcastle - would you like to review this?

File: reproject/mosaicking.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        x1, y1, x2, y2 = xr[order[0]], yr[order[0]], xr[order[1]], yr[order[1]]
+
+        # Determine angle between two of the vertices. It doesn't matter which
+        # ones they are, we just want to know how far from being straight the
+        # rectangle is.
+        angle = np.arctan2(y2 - y1, x2 - x1)
+
+        # Determine the smallest angle that would cause the rectangle to be
+        # lined up with the axes.
+        angle = angle % 90
```
Comment (keflavich) [inline reproject/mosaicking.py]: `np.arctan2` returns something in units of radians, not degrees, right?

Comment (keflavich) [reply reproject/mosaicking.py]: unless x1, x2, y1, and y2 are quantities?

Comment (astrofrog) [reply reproject/mosaicking.py]: Good catch!

Comment (astrofrog) [reply reproject/mosaicking.py]: They aren't quantities - this was a real bug

**Pull request merged**